diff --git a/build/conf/cross-build.inc b/build/conf/cross-build.inc
new file mode 100644
index 0000000..3ac5c74
--- /dev/null
+++ b/build/conf/cross-build.inc
@@ -0,0 +1,56 @@
+## for cross-build...
+# ament cmake
+IMAGE_INSTALL:append = " ament-cmake"
+IMAGE_INSTALL:append = " ament-cmake-auto"
+IMAGE_INSTALL:append = " ament-cmake-catch2"
+IMAGE_INSTALL:append = " ament-cmake-copyright"
+IMAGE_INSTALL:append = " ament-cmake-core"
+IMAGE_INSTALL:append = " ament-cmake-cppcheck"
+IMAGE_INSTALL:append = " ament-cmake-cpplint"
+IMAGE_INSTALL:append = " ament-cmake-export-definitions"
+IMAGE_INSTALL:append = " ament-cmake-export-dependencies"
+IMAGE_INSTALL:append = " ament-cmake-export-include-directories"
+IMAGE_INSTALL:append = " ament-cmake-export-interfaces"
+IMAGE_INSTALL:append = " ament-cmake-export-libraries"
+IMAGE_INSTALL:append = " ament-cmake-export-link-flags"
+IMAGE_INSTALL:append = " ament-cmake-export-targets"
+IMAGE_INSTALL:append = " ament-cmake-flake8"
+IMAGE_INSTALL:append = " ament-cmake-gmock"
+IMAGE_INSTALL:append = " ament-cmake-google-benchmark"
+IMAGE_INSTALL:append = " ament-cmake-gtest"
+IMAGE_INSTALL:append = " ament-cmake-include-directories"
+IMAGE_INSTALL:append = " ament-cmake-libraries"
+IMAGE_INSTALL:append = " ament-cmake-lint-cmake"
+IMAGE_INSTALL:append = " ament-cmake-mypy"
+IMAGE_INSTALL:append = " ament-cmake-nose"
+IMAGE_INSTALL:append = " ament-cmake-pclint"
+IMAGE_INSTALL:append = " ament-cmake-pep257"
+IMAGE_INSTALL:append = " ament-cmake-pycodestyle"
+IMAGE_INSTALL:append = " ament-cmake-pyflakes"
+IMAGE_INSTALL:append = " ament-cmake-pytest"
+IMAGE_INSTALL:append = " ament-cmake-python"
+IMAGE_INSTALL:append = " ament-cmake-ros"
+IMAGE_INSTALL:append = " ament-cmake-target-dependencies"
+IMAGE_INSTALL:append = " ament-cmake-test"
+IMAGE_INSTALL:append = " ament-cmake-uncrustify"
+IMAGE_INSTALL:append = " ament-cmake-version"
+IMAGE_INSTALL:append = " ament-cmake-xmllint"
+IMAGE_INSTALL:append = " ament-copyright"
+IMAGE_INSTALL:append = " ament-cppcheck"
+IMAGE_INSTALL:append = " ament-cpplint"
+IMAGE_INSTALL:append = " ament-flake8"
+IMAGE_INSTALL:append = " ament-index-cpp"
+IMAGE_INSTALL:append = " ament-index-python"
+IMAGE_INSTALL:append = " ament-lint"
+IMAGE_INSTALL:append = " ament-lint-auto"
+IMAGE_INSTALL:append = " ament-lint-cmake"
+IMAGE_INSTALL:append = " ament-lint-common"
+IMAGE_INSTALL:append = " ament-mypy"
+IMAGE_INSTALL:append = " ament-nodl"
+IMAGE_INSTALL:append = " ament-pclint"
+IMAGE_INSTALL:append = " ament-pep257"
+IMAGE_INSTALL:append = " ament-pycodestyle"
+IMAGE_INSTALL:append = " ament-pyflakes"
+IMAGE_INSTALL:append = " ament-uncrustify"
+IMAGE_INSTALL:append = " ament-xmllint"
+IMAGE_INSTALL:append = " ament-package"
diff --git a/build/conf/local.conf b/build/conf/local.conf
index 1827217..ee96df7 100644
--- a/build/conf/local.conf
+++ b/build/conf/local.conf
@@ -290,7 +290,7 @@ DISTRO_FEATURES_NATIVESDK_remove = " x11"
 
 CIP_MODE = "Buster"
 
-INCOMPATIBLE_LICENSE = "GPLv3 GPLv3+"
+#INCOMPATIBLE_LICENSE = "GPLv3 GPLv3+"
 
 LICENSE_FLAGS_WHITELIST = " \
 	commercial_gstreamer1.0-plugins-ugly commercial_mpeg2dec \
@@ -339,3 +339,61 @@ IMAGE_ROOTFS_EXTRA_SPACE = "1048576"
 #SPDX_ARCHIVE_PACKAGED = "1"
 
 IMAGE_INSTALL_append = " tesseract tesseract-lang "
+
+######
+IMAGE_INSTALL:append = " opencv opencv-dev"
+PACKAGECONFIG_append_pn-opencv = " gtk"
+PACKAGECONFIG_append_pn-nativesdk-opencv = " gtk"
+IMAGE_INSTALL:append = " sudo"
+
+IMAGE_INSTALL:append = " libgomp libgomp-dev libgomp-staticdev"
+
+# ros base
+#IMAGE_INSTALL_append = " packagegroup-ros-world"
+IMAGE_INSTALL:append = " ros-core"
+IMAGE_INSTALL:append = " demo-nodes-cpp"
+
+# additional ros2 pkg. (moveit2 depend's packages etc.)
+include conf/other.inc
+
+# V4L
+IMAGE_INSTALL:append = " v4l2-init v4l-utils"
+IMAGE_INSTALL:append = " v4l2-camera"
+
+# tf/tf2
+include conf/tf.inc
+
+# realsense
+include conf/realsense2.inc
+
+# image-transport-plugins
+IMAGE_INSTALL:append = " image-transport-plugins"
+
+# for cross-build...
+include conf/cross-build.inc
+
+# rmw
+include conf/rmw.inc
+
+# rcl
+include conf/rcl.inc
+
+# LiDAR node
+IMAGE_INSTALL:append = " hls-lfcd-lds-driver"
+
+# navigation2
+include conf/navigation2.inc
+
+# cartographer-ros
+IMAGE_INSTALL:append = " cartographer-ros"
+
+# rtabmap
+PNBLACKLIST[rtabmap]=""
+IMAGE_INSTALL:append = " rtabmap"
+IMAGE_INSTALL:append = " rtabmap-ros"
+
+# Measures to take when an error occurs "ninja: build stopped: subcommand failed."
+#BB_NUMBER_THREADS = '1'
+#PARALLEL_MAKE = '-j 1'
+
+INHERIT += "rm_work"
diff --git a/build/conf/navigation2.inc b/build/conf/navigation2.inc
new file mode 100644
index 0000000..53ce14c
--- /dev/null
+++ b/build/conf/navigation2.inc
@@ -0,0 +1,9 @@
+PNBLACKLIST[navigation2] = ""
+PNBLACKLIST[nav2-bringup] = ""
+IMAGE_INSTALL:append = " navigation2 nav2-bringup"
+
+## X11 not support...
+RDEPENDS:${PN}:remove = " nav2-rviz-plugins"
+
+# OPEN-MP not support...
+RDEPENDS:${PN}:remove = " nav2-smac-planner"
diff --git a/build/conf/other.inc b/build/conf/other.inc
new file mode 100644
index 0000000..c50ce16
--- /dev/null
+++ b/build/conf/other.inc
@@ -0,0 +1,17 @@
+# additional ros2 pkg. (moveit2 depend's packages etc.)
+IMAGE_INSTALL:append = " control-msgs"
+IMAGE_INSTALL:append = " geometry2"
+IMAGE_INSTALL:append = " eigen-stl-containers"
+IMAGE_INSTALL:append = " octomap-msgs"
+IMAGE_INSTALL:append = " geometric-shapes"
+IMAGE_INSTALL:append = " object-recognition-msgs"
+IMAGE_INSTALL:append = " random-numbers"
+IMAGE_INSTALL:append = " srdfdom"
+#IMAGE_INSTALL:append = " urdf-parser-py"
+IMAGE_INSTALL:append = " urdf-parser-plugin"
+IMAGE_INSTALL:append = " urdfdom-py"
+IMAGE_INSTALL:append = " octomap"
+IMAGE_INSTALL:append = " eigen3-cmake-module"
+IMAGE_INSTALL:append = " image-geometry"
+
+IMAGE_INSTALL:append = " ros2-control"
diff --git a/build/conf/rcl.inc b/build/conf/rcl.inc
new file mode 100644
index 0000000..6fc532a
--- /dev/null
+++ b/build/conf/rcl.inc
@@ -0,0 +1,13 @@
+# rcl
+IMAGE_INSTALL:append = " rcl"
+IMAGE_INSTALL:append = " rcl-action"
+IMAGE_INSTALL:append = " rcl-interfaces"
+IMAGE_INSTALL:append = " rcl-lifecycle"
+IMAGE_INSTALL:append = " rcl-logging-interface"
+IMAGE_INSTALL:append = " rcl-logging-log4cxx"
+IMAGE_INSTALL:append = " rcl-logging-noop"
+IMAGE_INSTALL:append = " rcl-logging-spdlog"
+IMAGE_INSTALL:append = " rcl-yaml-param-parser"
+IMAGE_INSTALL:append = " rcl-yaml-param-parser"
+
+IMAGE_INSTALL:append = " python3-numpy"
diff --git a/build/conf/realsense2.inc b/build/conf/realsense2.inc
new file mode 100644
index 0000000..f98e353
--- /dev/null
+++ b/build/conf/realsense2.inc
@@ -0,0 +1,6 @@
+# realsense
+PNBLACKLIST[librealsense2] = ""
+PNBLACKLIST[realsense2-camera] = ""
+IMAGE_INSTALL:append = " librealsense2"
+IMAGE_INSTALL:append = " realsense2-camera"
+IMAGE_INSTALL:append = " realsense2-camera-msgs"
diff --git a/build/conf/rmw.inc b/build/conf/rmw.inc
new file mode 100644
index 0000000..d5ee188
--- /dev/null
+++ b/build/conf/rmw.inc
@@ -0,0 +1,17 @@
+# rmw
+IMAGE_INSTALL:append = " rmw"
+IMAGE_INSTALL:append = " rmw-implementation"
+IMAGE_INSTALL:append = " rmw-implementation-cmake"
+IMAGE_INSTALL:append = " rmw-cyclonedds-cpp"
+IMAGE_INSTALL:append = " rmw-dds-common"
+IMAGE_INSTALL:append = " rmw-fastrtps-cpp"
+IMAGE_INSTALL:append = " rmw-fastrtps-dynamic-cpp"
+IMAGE_INSTALL:append = " rmw-fastrtps-shared-cpp"
+
+IMAGE_INSTALL:append = " fastcdr"
+IMAGE_INSTALL:append = " fastrtps"
+IMAGE_INSTALL:append = " fastrtps-cmake-module"
+
+IMAGE_INSTALL:append = " libyaml libyaml-dev"
+IMAGE_INSTALL:append = " libyaml-vendor libyaml-vendor-dev"
+IMAGE_INSTALL:append = " foonathan-memory foonathan-memory-vendor"
diff --git a/build/conf/tf.inc b/build/conf/tf.inc
new file mode 100644
index 0000000..f52a633
--- /dev/null
+++ b/build/conf/tf.inc
@@ -0,0 +1,5 @@
+# tf/tf2
+IMAGE_INSTALL:append = " tf-transformations"
+IMAGE_INSTALL:append = " tf2 tf2-bullet tf2-eigen tf2-eigen-kdl"
+IMAGE_INSTALL:append = " tf2-geometry-msgs tf2-kdl tf2-msgs tf2-py"
+IMAGE_INSTALL:append = " tf2-ros tf2-ros-py tf2-sensor-msgs"
